### PR TITLE
Add ability to specify custom maven repositories

### PR DESCRIPTION
**Pull request by nicolaferraro**
Fixes ENTESB-11264

Custom repositories can be now specified as in:
```
apiVersion: syndesis.io/v1alpha1
kind: Syndesis
metadata:
  name: app
spec:
  mavenRepositories:
    another: https://yyy.xxx/v2/
    myrepo: http://xxx.yyy/path
  # ...
```

They are picked up by both classic and camel k runtimes.

### CODE REVIEW

File: install/operator/pkg/generator/assets/infrastructure/03-syndesis-server-config.yml.tmpl [hunk 1]
```diff
@@ -64,17 +64,25 @@
...
+      maven:
+        repositories:
+    {{- range $id, $url := .Syndesis.Spec.MavenRepositories}}
+          {{ $id }}: {{ $url }}
+    {{- end }}
+{{- else}}
+    {{- if .Productized}}
       maven:
         repositories:
           central: https://repo.maven.apache.org/maven2/
```
Comment (lgarciaaco) [inline install/operator/pkg/generator/assets/infrastructure/03-syndesis-server-config.yml.tmpl]: Why don't we rather set a default value for `.Syndesis.Spec.MavenRepositories` so we dont have to ask every time whether is has a value or not?

Comment (nicolaferraro) [reply install/operator/pkg/generator/assets/infrastructure/03-syndesis-server-config.yml.tmpl]: That's a better solution. Iirc default values handling is still work in progress, isn't it?

Comment (lgarciaaco) [reply install/operator/pkg/generator/assets/infrastructure/03-syndesis-server-config.yml.tmpl]: We have this section starting here, it seems the right place to do it: https://github.com/syndesisio/syndesis/blob/9fab4e22937b38a7e18bbadbc684aa1c04488689/install/operator/pkg/syndesis/template/install.go#L146-L151

**Comment by zregvart** [comment]
@lgarciaaco @nicolaferraro do we need to do something else here other than a rebase?

**Comment by nicolaferraro** [comment]
@zregvart I'm working on the fix suggested by @lgarciaaco.

**Comment by nicolaferraro** [comment]
@lgarciaaco can you have another look now?

**Comment by nicolaferraro** [comment]
Any hint on how to unlock the "syndesis" check?

**Comment by zregvart** [comment]
> Any hint on how to unlock the "syndesis" check?

Seems that CircleCI has made some changes to the checks, we're no longer getting the cumulative checks as `Syndesis` but each job as separate check, I've disabled mandatory `Syndesis` check for `master`...

**Pull request merged**